glean-core/android/src/main/java/mozilla/telemetry/glean/private/EventMetricType.kt

```diff
@@ -117,15 +142,39 @@ class EventMetricType<ExtraKeysEnum : Enum<ExtraKeysEnum>>(
      */
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testHasValue(pingName: String = sendInPings.first()): Boolean {
-        // @Suppress("EXPERIMENTAL_API_USAGE")
-        // Dispatchers.API.assertInTestingMode()
-
-        // val snapshot = EventsStorageEngine.getSnapshot(pingName, false) ?: return false
-        // return snapshot.any { event ->
-        //     event.identifier == identifier
-        // }
-        assert(false, { "Testing API not implementated for EventMetricType" })
-        return false
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        return LibGleanFFI.INSTANCE.glean_event_test_has_value(
+            Glean.handle,
+            this.handle,
+            pingName
+        ).toBoolean()
+    }
+
+    /**
+    * Deserializes an event in JSON into a RecordedEventData object.
+    *
+    * @param jsonContent The JSONObject containing the data for the event.
```

Comment:
nit: can you document the expected structure of the JSONobject?

Reply:
It's the same as the event in a data ping, but it doesn't hurt to document it here.